src/dependency-manager/src/manager.ts

```diff
@@ -76,16 +77,32 @@ export default abstract class DependencyManager {
     }
 
     const gateway_node = this.graph.nodes.find((node) => (node instanceof GatewayNode));
-    const gateway_port = gateway_node ? gateway_node.ports.expose.toString() : '80';
+    const gateway_port = gateway_node ? gateway_node.ports[0].expose.toString() : '80';
     for (const node of this.graph.nodes) {
       const external_host = subdomain_map[node.ref] ? `${subdomain_map[node.ref]}.localhost` : '';
       const internal_host = node instanceof ExternalNode ? node.host : node.normalized_ref;
+
       env_params_to_expand[`${node.normalized_ref.toUpperCase()}_EXTERNAL_HOST`.replace(/[.-]/g, '_')] = external_host;
-      env_params_to_expand[`${node.normalized_ref.toUpperCase()}_INTERNAL_HOST`.replace(/[.-]/g, '_')] = internal_host;
-      env_params_to_expand[`${node.normalized_ref.toUpperCase()}_HOST`.replace(/[.-]/g, '_')] = external_host ? external_host : internal_host;
+      if (internal_host) {
+        env_params_to_expand[`${node.normalized_ref.toUpperCase()}_INTERNAL_HOST`.replace(/[.-]/g, '_')] = internal_host;
```

Comment:
Can you add a helper method that takes a node and a string to transform these values? The # of replaces is giving me hives. 
```
`${node.normalized_ref.toUpperCase()}_EXTERNAL_HOST`.replace(/[.-]/g, '_')
```
something like:
`scopeEnv(node, 'EXTERNAL_HOST')`

Reply:
Absolutely